src/wp-includes/blocks.php

```diff
@@ -1009,6 +1009,24 @@ function excerpt_remove_blocks( $content ) {
 	return $output;
 }
 
+/**
+ * Parses footnotes markup out of a content string,
+ * and renders those appropriate for the excerpt.
+ *
+ * @since 6.3.0
+ *
+ * @param string $content The content to parse.
+ * @return string The parsed and filtered content.
+ */
+function excerpt_remove_footnotes( $content ) {
+	if ( false === strpos( $content, 'data-fn=' ) ) {
```

Comment:
```suggestion
	if ( ! str_contains( $content, 'data-fn=' ) ) {
```

Per [#58206](https://core.trac.wordpress.org/ticket/58206), Use str_contains instead strpos.